plugins/filter/time.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, René Moser <mail@renemoser.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+from ansible.errors import AnsibleFilterError
+
+
+UNIT_FACTORS = {
+    'ms': [],
+    's': [1000],
+    'm': [1000, 60],
+    'h': [1000, 60, 60],
+    'd': [1000, 60, 60, 24],
+    'w': [1000, 60, 60, 24, 7],
+    'y': [1000, 60, 60, 24, 365],
+}
+
+
+def multiply(factors):
+    result = 1
+    for factor in factors:
+         result = result * factor
+    return result
+
+
+def divide(divisors):
+    result = 1
+    for divisor in divisors:
+         result = result / divisor
+    return result
+
+
+def to_time_unit(human_time, unit='ms'):
+    ''' Return a time unit from a human readable string '''
+    if unit not in UNIT_FACTORS:
+        raise AnsibleFilterError("to_time_unit() can not convert to the following unit: %s, "
+                                 "available units: %s" % (unit, ', '.join(UNIT_FACTORS.keys())))
+    result = 0
+    for h_time_string in human_time.split():
+        res = re.match(r'(-?\d+)(\w+)', h_time_string)
+        if not res:
+            raise AnsibleFilterError(
+                "to_time_unit() can not interpret following string: %s" % human_time)
+
+        h_time_int = int(res.group(1))
+        h_time_unit = res.group(2)
+
+        if h_time_unit not in UNIT_FACTORS:
+            raise AnsibleFilterError(
+                "to_time_unit() can not interpret following string: %s" % human_time)
+
+        time_in_milliseconds = h_time_int * multiply(UNIT_FACTORS[h_time_unit])
+        result += time_in_milliseconds * divide(UNIT_FACTORS[unit])
+    return round(result, 12)
```

Comment:
How about:
```suggestion
        result += time_in_milliseconds
    return round(result * divide(UNIT_FACTORS[unit]), 12)
```
This reduces potential rounding errors (as computations are generally made as `int`, not as `float`).

Also, you could replace `result * divide(...)` by `result / multiply(...)` :)